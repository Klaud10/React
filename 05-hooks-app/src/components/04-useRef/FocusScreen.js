import React, { useRef } from 'react';
import './useReef.css';

export const FocusScreen = () => {

    const inputRef = useRef();

    const  handleClick = () => {
        inputRef.current.select();
        console.log(inputRef);
    };  

    return (
        <div className="mb-3">
            <h1>Focus Screen</h1>
            
            <label className="form-label"></label>
            
            <input  
                ref={ inputRef }
                className="form-control mb-3"
                placeholder="Nombre"
            />

            <button className="btn btn-outline-primary"
                    onClick={ handleClick }
            >
                Focus
            </button>
            
        </div>
    )
}
