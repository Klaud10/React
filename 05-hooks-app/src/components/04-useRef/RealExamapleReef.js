import React, {useState} from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RealExamapleReef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Real Reef</h1>
            <hr />

            { show && <MultipleCustomHooks />}

            <button
                className="btn btn-success mt-5"
                onClick={ () => { setShow ( !show )}}
            >
                Mostrar/Ocultar
            </button>
        </div>
    )
}
