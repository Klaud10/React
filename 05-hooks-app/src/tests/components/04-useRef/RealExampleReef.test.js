import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleReef';

describe('Pruebas en <RealExampleRef />', () => {
    
    const wrapper = shallow( <RealExampleRef /> );

    test('debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);

    });
    
    test('Debe mostrar el componete <MultipleCustomHooks />', () => {
        
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
    });
});
