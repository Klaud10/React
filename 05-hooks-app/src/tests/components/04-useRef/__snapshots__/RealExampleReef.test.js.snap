// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Reef
  </h1>
  <hr />
  <button
    className="btn btn-success mt-5"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</div>
`;
